torch/lib/c10d/ProcessGroupMPI.cpp

```diff
@@ -150,8 +150,11 @@ ProcessGroupMPI::AsyncWork::AsyncWork(
 
 ProcessGroupMPI::AsyncWork::~AsyncWork() {
   if (request_ != MPI_REQUEST_NULL) {
-    throw std::runtime_error(
-        "Attempted destruction of AsyncWork before work has completed");
+    std::cerr
+        << "Attempted destruction of AsyncWork before work has completed, "
+        << "terminating the program."
+        << std::endl;
```

Comment:
Please add a std::flush as well to be sure it is written to the underlying STDERR

Reply:
std::endl will call os.flush(). Adding one more CI change before landing